### PR TITLE
Release version 0.5.0 with automation

### PR DESCRIPTION
**Pull request by dcarley**
So that we don't need to find credentials and remember how to release
changes manually. Git tags are used to trigger publishing so that we
don't re-publish over existing versions every time we merge to master.
It also means that we're more likely to remember to tag and don't need
to add any logic or credentials for the workflow to create tags.

An alternative would have been to use the build number as a patch
version, but it's unwieldy for both for users and for us to maintain the
versions in `README.md` and `CHANGELOG.md`.

The `CLOJARS_USERNAME` and `CLOJARS_TOKEN` environment variables have
been added to the project. At the moment they belong to an individual
but we'll move them to a machine user later. The project already has
"build forks" and "pass secrets to fork builds" disabled, which will
prevent leaking these credentials.

---

Depends on #53 and #54.

### CODE REVIEW

**Comment by calvis** [comment]
tests failed because this is using an ancient version of clojure `[org.clojure/clojure "1.7.0"]`

**Comment by aterweele** [comment]
> tests failed because this is using an ancient version of clojure `[org.clojure/clojure "1.7.0"]`

@calvis I have a fix for that in 8946a91d59f606ff6b9ecb3aa7893a81b6311c1c, so merging #51 and rebasing this PR ought to fix that

**Comment by calvis** [comment]
can we not update the Clojure version, instead?

**Comment by dcarley** [comment]
> can we not update the Clojure version, instead?

There be dragons, it turns out. I've raised #53 and #54 to take another approach. I'll rebase this PR if they're merged.

**Comment by aterweele** [comment]
I'm going to bow out of this and leave it up to y'all what should be done here. Thanks for your work to cut a release!

**Comment by dcarley** [comment]
I've rebased this to no longer use build number for patch versions, added a CHANGELOG, and bumped for version 0.5.0

File: project.clj [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
   :dependencies []
   :plugins [[lein-cloverage "1.2.2"]]
-  :profiles {:dev {:dependencies [[org.clojure/clojure "1.10.2"]]}})
+  :profiles {:dev {:dependencies [[org.clojure/clojure "1.10.2"]]}}
+  :repositories [["releases" {:url "https://clojars.org/repo"
+                              :username :env/clojars_username
+                              :password :env/clojars_token
+                              :sign-releases false}]
```
Comment (skaasten) [inline project.clj:11]: Curious what's needed for us to use signed releases.

**Review by skaasten — APPROVED** [review]
Using tags makes sense to me - nice to have this semi-automated.

File: project.clj (hunk 1 above)
Comment (c2nes) [reply project.clj:11]: If it's anything like how Maven does it then we'd need to publish a PGP key tied to CircleCI (e.g. `oss@circleci.com`) and then set up the necessary keychain in CI. The most painful part in my experience is finding a good process for managing and rotating the PGP key.

Comment (c2nes) [reply project.clj:11]: and IIRC signatures aren't validated by Maven by default so the valuing in signing these would just be to folks who choose to enable validation

**Pull request merged**